File: src/wwwroot/app/common/models/business/offer/createOfferCommand.ts
/// <reference path="../../../../typings/_all.d.ts" />

module Antares.Common.Models.Business {
    export class CreateOfferCommand {
        activityId: string;
        requirementId: string;
        offerTypeId: string;
        statusId: string;
        price: number;
        pricePerWeek: number;
        offerDate: Date;
        exchangeDate: Date;
        completionDate: Date;
        specialConditions: string;

        constructor(model?: CreateOfferCommand){
            model = model || <CreateOfferCommand>{};
            this.activityId = model.activityId;
            this.requirementId = model.requirementId;
            this.offerTypeId = model.offerTypeId;
            this.statusId = model.statusId;
            this.price = model.price;
            this.pricePerWeek = model.pricePerWeek;
            this.offerDate = model.offerDate;
            this.exchangeDate = model.exchangeDate;
            this.completionDate = model.completionDate;
            this.specialConditions = model.specialConditions;
        }
    }
}

File: src/KnightFrank.Antares.Domain/Offer/Commands/CreateOfferCommand.cs
namespace KnightFrank.Antares.Domain.Offer.Commands
{
    using System;

    using MediatR;

    public class CreateOfferCommand : IRequest<Guid>
    {
        public Guid ActivityId { get; set; }

        public Guid RequirementId { get; set; }

        public Guid StatusId { get; set; }

        public decimal? Price { get; set; }

        public decimal? PricePerWeek { get; set; }

        public DateTime OfferDate { get; set; }

        public DateTime? ExchangeDate { get; set; }

        public DateTime? CompletionDate { get; set; }

        public string SpecialConditions { get; set; }
    }
}
